kyo-core/shared/src/test/scala/kyo/Test.scala

```diff
@@ -16,13 +16,16 @@ abstract class Test extends AsyncFreeSpec with BaseKyoTest[Abort[Any] & Async &
 
     def run(v: Future[Assertion] < (Abort[Any] & Async & Resource)): Future[Assertion] =
         import AllowUnsafe.embrace.danger
-        val a = Async.run(Abort.run(Resource.run(v)).map(_.fold {
-            _.getFailure match
+        v.pipe(
+            Resource.run,
+            Abort.recover[Any] {
                 case ex: Throwable => throw ex
                 case e             => throw new IllegalStateException(s"Test aborted with $e")
```

Comment:
The compiler started complaining of an unreachable case here but it's incorrect. I tried to reproduce in isolation but it seems bad interaction with inlining. As a workaround, I've rewrote the method to use `pipe`, which fixed the warning and seems more readable.